internal/xds/translator/jsonpatch.go

```diff
@@ -240,125 +240,150 @@ func processJSONPatches(tCtx *types.ResourceVersionTable, envoyPatchPolicies []*
 				}
 			}
 
-			// Convert patch to JSON
-			// The patch library expects an array so convert it into one
-			y, err := yaml.Marshal([]ir.JSONPatchOperation{p.Operation})
-			if err != nil {
-				tErr := fmt.Errorf("unable to marshal patch %+v, err: %s", p.Operation, err.Error())
-				tErrs = errors.Join(tErrs, tErr)
-				continue
-			}
-			jsonBytes, err := yaml.YAMLToJSON(y)
-			if err != nil {
-				tErr := fmt.Errorf("unable to convert patch to json %s, err: %s", string(y), err.Error())
-				tErrs = errors.Join(tErrs, tErr)
-				continue
-			}
-			patchObj, err := jsonpatchv5.DecodePatch(jsonBytes)
-			if err != nil {
-				tErr := fmt.Errorf("unable to decode patch %s, err: %s", string(jsonBytes), err.Error())
-				tErrs = errors.Join(tErrs, tErr)
-				continue
-			}
-
-			// Apply patch
-			opts := jsonpatchv5.NewApplyOptions()
-			opts.EnsurePathExistsOnAdd = true
-			modifiedJSON, err := patchObj.ApplyWithOptions(resourceJSON, opts)
-			if err != nil {
-				tErr := fmt.Errorf("unable to apply patch:\n%s on resource:\n%s, err: %s", string(jsonBytes), string(resourceJSON), err.Error())
-				tErrs = errors.Join(tErrs, tErr)
-				continue
-			}
-
-			// Unmarshal back to typed resource
-			// Use a temp staging variable that can be marshalled
-			// into and validated before saving it into the xds output resource
-			switch p.Type {
-			case resourcev3.ListenerType:
-				temp := &listenerv3.Listener{}
-				if err = protojson.Unmarshal(modifiedJSON, temp); err != nil {
-					tErr := fmt.Errorf(unmarshalErrorMessage(err, string(modifiedJSON)))
-					tErrs = errors.Join(tErrs, tErr)
-					continue
+			var jsonPointers []string
+			if p.Operation.JSONPath != nil {
```

Comment:
are we expecting JSONPath and Path to be set at the same time ?

Reply:
looks like JSONPath supports nested values
https://datatracker.ietf.org/doc/rfc9535/

Reply:
@arkodg yes, both properties can be set at the same time, because JSONPath is only able to select what already exists in the JsonDocument. 

If one want to create a new property at a certain element, then this can be done by:
1. Select the existing element with `JSONPath` Property
2. `Path` Property is added to the selected existing element
3. Element can be created at a not existing path